samcli/lib/utils/async_utils.py

```diff
@@ -113,4 +120,12 @@ def run_async(self):
         List of result of the executions in order
         """
         event_loop = asyncio.new_event_loop()
-        return run_given_tasks_async(self._async_tasks, event_loop)
+        with ThreadPoolExecutor() as self.executor:
+            return run_given_tasks_async(self._async_tasks, event_loop, self.executor)
```

Comment:
Can we make creation of `self.executor` optional?

We can add a param to constructor or the `run_async` method which will be a flag (True/False), and it will set `self.executor` to new `ThreadPoolExecutor` instance. If it is not provided it will use `None` as it was using before.

Reply:
done